rclpy/test/test_node.py

```diff
@@ -991,6 +991,83 @@ def test_node_get_parameters_by_prefix(self):
         self.assertFalse(parameters)
         self.assertIsInstance(parameters, dict)
 
+    def test_node_list_parameters(self):
+        parameters = [
+            ('foo_prefix.foo', 43),
+            ('foo_prefix.bar', 'hello'),
+            ('foo_prefix.baz', 2.41),
+            ('bar_prefix.foo', 1),
+            ('bar_prefix.bar', 12.3),
+            ('bar_prefix.baz', 'world'),
+        ]
+        self.node.declare_parameters('', parameters)
+
+        parameter_names = self.node.list_parameters()
+        self.assertIsInstance(parameter_names, list)
+        self.assertEqual(len(parameter_names), 7)
+        self.assertCountEqual(
+            parameter_names,
+            [
+                'foo_prefix.foo',
+                'foo_prefix.bar',
+                'foo_prefix.baz',
+                'bar_prefix.foo',
+                'bar_prefix.bar',
+                'bar_prefix.baz',
+                USE_SIM_TIME_NAME
+            ]
+        )
+
+        parameter_names = self.node.list_parameters(
+            prefixes=['foo_prefix', 'bar_prefix'])
+        self.assertIsInstance(parameter_names, list)
+        self.assertEqual(len(parameter_names), 6)
+        self.assertCountEqual(
+            parameter_names,
+            [
+                'foo_prefix.foo',
+                'foo_prefix.bar',
+                'foo_prefix.baz',
+                'bar_prefix.foo',
+                'bar_prefix.bar',
+                'bar_prefix.baz'
+            ]
+        )
+
+        parameter_names = self.node.list_parameters(
+            prefixes=['foo_prefix'])
```

Comment:
how about adding the test case that prefix includes `.`? like `prefixes=['foo_prefix.']`

Reply:
Thanks for the good test case. i think a prefix including `.` also be a valid option. right?
For this, i replaced the line
`if name.startswith(prefix + PARAMETER_SEPARATOR_STRING):` with 
`if parameter_name.startswith(prefix):`

I noticed that the current implementation in `parameter_service.py` still uses
the old line. I would be happy to update it if my proposed change is indeed correct.